app/src/main/java/com/hedvig/app/feature/checkout/EditQuotesUseCase.kt

```diff
@@ -0,0 +1,123 @@
+import com.hedvig.app.util.apollo.safeQuery
+import com.hedvig.app.util.featureflags.Feature
+import com.hedvig.app.util.featureflags.FeatureManager
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.put
+import kotlinx.serialization.json.putJsonObject
+import org.json.JSONException
+import org.json.JSONObject
+
+class EditQuotesUseCase(
+    private val apolloClient: ApolloClient,
+    private val featureManager: FeatureManager,
+    private val localeManager: LocaleManager,
+    private val graphQLQueryHandler: GraphQLQueryHandler,
+) {
+
+    data class Error(val message: String? = null)
+
+    object Success
+
+    suspend fun editQuotes(parameter: EditAndSignParameter): Either<Error, Success> = parameter.quoteIds
+        .map { editQuote(it, parameter.quoteCartId, parameter.ssn, parameter.email) }
+        .sequenceEither()
+        .map { Success }
+
+    private suspend fun editQuote(
+        quoteId: String,
+        quoteCartId: String?,
+        ssn: String,
+        email: String
+    ): Either<Error, Success> {
+        return if (featureManager.isFeatureEnabled(Feature.QUOTE_CART)) {
+            if (quoteCartId == null) {
+                Either.Left(Error("No quote cart id found"))
+            } else {
+                editQuoteCart(quoteCartId, quoteId, ssn, email)
+            }
+        } else {
+            editQuoteWithEmailAndSSN(quoteId, ssn, email)
+        }
+    }
+
+    private suspend fun editQuoteCart(
+        quoteCartId: String,
+        quoteId: String,
+        ssn: String,
+        email: String
+    ): Either<Error, Success> {
+        val json = buildJsonObject {
+            put("quoteCartId", quoteCartId)
+            put("quoteId", quoteId)
+            put("locale", localeManager.defaultLocale().rawValue)
+            putJsonObject("payload") {
+                put("ssn", ssn)
+                put("email", email)
+                putJsonObject("data") {
+                    put("ssn", ssn)
+                    put("email", email)
+                }
+            }
+        }.toString()
+
+        return graphQLQueryHandler.graphQLQuery(
+            query = QuoteCartEditQuoteMutation.QUERY_DOCUMENT,
+            variables = JSONObject(json),
+            files = emptyList()
+        )
+            .toEither()
+            .mapLeft { Error() }
+            .flatMap { jsonResponse ->
+                val errorCode = try {
+                    jsonResponse.getJSONObject("data")
+                        .getJSONObject("quoteCart_editQuote")
+                        .getJSONArray("limits")
+                        .getJSONObject(0)
+                        .getString("code")
+                } catch (exception: JSONException) {
+                    null
+                }
+
+                if (errorCode != null) {
+                    Either.Left(Error(errorCode))
+                } else {
+                    Either.Right(Success)
+                }
```

Comment:
I think these can also be written as `Error(errorCode).left()` and `Success.right()` respectively. I usually like avoiding having to write`Either.Right(` for example. Just something you may want to know if you haven't stumbled upon it yet.